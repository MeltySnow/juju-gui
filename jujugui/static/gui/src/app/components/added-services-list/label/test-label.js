/* Copyright (C) 2018 Canonical Ltd. */
'use strict';

const enzyme = require('enzyme');
const React = require('react');

const AddedServicesLabel = require('./label');

describe('AddedServicesLabel', () => {

  const bundleURL = 'elasticsearch-cluster/bundle/17';

  const renderComponent = (options = {}) => enzyme.shallow(
    <AddedServicesLabel
      bundleURL={options.bundleURL || bundleURL}
      changeState={options.changeState || sinon.stub()} />
  );

  it('renders', () => {
    const component = renderComponent();
    expect(component).toMatchSnapshot();
  });

  it('calls to show readme on click', () => {
    const changeState = sinon.stub();
    const component = renderComponent({changeState});
    component.find('.inspector-view__label-link-list li').at(0).simulate('click');
    assert.equal(changeState.callCount, 1);
    assert.deepEqual(changeState.args[0][0], {store: bundleURL});
  });

  it('calls to show post deployment on click', () => {
    const changeState = sinon.stub();
    const component = renderComponent({changeState});
    component.find('.inspector-view__label-link-list li').at(1).simulate('click');
    assert.equal(changeState.callCount, 1);
    assert.deepEqual(changeState.args[0][0], {postDeploymentPanel: bundleURL});
  });

});
