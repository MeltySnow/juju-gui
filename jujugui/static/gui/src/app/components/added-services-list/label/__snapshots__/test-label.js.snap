// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddedServicesLabel renders 1`] = `
<li
  className="inspector-view__label-item"
>
  <div
    className="inspector-view__label-name"
  >
    elasticsearch cluster
  </div>
  <ul
    className="inspector-view__label-link-list"
  >
    <li
      onClick={[Function]}
    >
      Readme
    </li>
    <li
      onClick={[Function]}
    >
      Get Started
    </li>
  </ul>
</li>
`;
